index.js

```diff
@@ -480,16 +483,51 @@ export default function(olLayer, glStyle, source, resolutions, spriteData, sprit
                   style = styles[stylesLength] = new Style();
                 }
                 style.setGeometry(styleGeom);
-                iconImg = iconImageCache[icon];
+                var iconSize = paint['icon-size'](zoom, properties);
+                var iconColor = paint['icon-color'](zoom, properties);
+                var icon_cache_key = icon + '.' + iconSize + '.' + iconColor;
+                iconImg = iconImageCache[icon_cache_key];
                 if (!iconImg) {
                   var spriteImageData = spriteData[icon];
-                  iconImg = iconImageCache[icon] = new Icon({
-                    img: spriteImage,
-                    imgSize: spriteImgSize,
-                    size: [spriteImageData.width, spriteImageData.height],
-                    offset: [spriteImageData.x, spriteImageData.y],
-                    scale: paint['icon-size'](zoom, properties) / spriteImageData.pixelRatio
-                  });
+                  if (iconColor[0] !== 0 || iconColor[1] !== 0 || iconColor[2] !== 0) {
```

Comment:
This would be more efficient if there was no default for `icon-color`.

Reply:
agreed, I took the default from the mapbox style spec, but indeed it makes more sense to have no default, will change

Reply:
Right. A default would make sense for sdf icons, but we don't have those, and png icons have their own color already.

Reply:
See: https://github.com/boundlessgeo/mapbox-to-ol-style/pull/20/commits/cde172bd69750a661ccf84da541a77f150b48f1e